lib/saluki-config/src/refresher.rs

```diff
@@ -0,0 +1,131 @@
+
+    /// The amount of time betweeen each request in seconds.
+    ///
+    /// Defaults to 15 seconds.
+    #[serde(default = "default_refresh_interval_seconds")]
+    refresh_interval_seconds: u64,
+}
+
+fn default_auth_token_file_path() -> String {
+    DEFAULT_AUTH_TOKEN_FILE_PATH.to_owned()
+}
+
+fn default_refresh_interval_seconds() -> u64 {
+    DEFAULT_REFRESH_INTERVAL_SECONDS
+}
+
+/// A configuration whose values are refreshed from a remote source at runtime.
+#[derive(Clone, Default)]
+pub struct RefreshableConfiguration {
+    token: String,
+    values: Arc<ArcSwap<Value>>,
+    refresh_interval_seconds: u64,
+}
+
+impl RefresherConfiguration {
+    /// Creates a new `RefresherConfiguration` from the given configuration.
+    pub fn from_configuration(config: &GenericConfiguration) -> Result<Self, GenericError> {
+        Ok(config.as_typed()?)
+    }
+    /// Create `RefreshableConfiguration` from `RefresherConfiguration`.
+    pub fn build(&self) -> Result<Arc<RefreshableConfiguration>, GenericError> {
+        let raw_bearer_token = std::fs::read_to_string(&self.auth_token_file_path)?;
+        let refreshable_configuration = Arc::new(RefreshableConfiguration {
+            token: raw_bearer_token,
+            values: Arc::new(ArcSwap::from_pointee(serde_json::Value::Null)),
+            refresh_interval_seconds: self.refresh_interval_seconds,
+        });
+        refreshable_configuration.clone().spawn_refresh_task();
+        Ok(refreshable_configuration)
+    }
+}
+impl RefreshableConfiguration {
+    /// Start a task that queries the datadog-agent config endpoint every 15 seconds.
+    fn spawn_refresh_task(self: Arc<Self>) {
+        tokio::spawn(async move {
+            let client = ClientBuilder::new()
+                .danger_accept_invalid_certs(true) // Allow invalid certificates
+                .build()
+                .expect("failed to create http client");
+            loop {
+                let response = client
+                    .get(DATADOG_AGENT_CONFIG_ENDPOINT)
+                    .header("Content-Type", "application/json")
+                    .header("Authorization", format!("Bearer {}", self.token))
+                    .header("DD-Agent-Version", "0.1.0")
+                    .header("User-Agent", "agent-data-plane/0.1.0")
+                    .send()
+                    .await;
+                match response {
+                    Ok(response) => {
+                        let config_response: Value = response
+                            .json()
+                            .await
+                            .expect("failed to deserialize configuration into json");
+                        self.values.store(Arc::new(config_response));
+                        debug!("Retrieved configuration from datadog-agent.");
+                    }
+                    Err(e) => {
+                        error!("Error retrieving configuration from datadog-agent: {}", e);
```

Comment:
I would change this just to be slightly more generic:

```suggestion
                        error!(remote_endpoint = DATADOG_AGENT_CONFIG_ENDPOINT, "Failed to retrieve configuration from remote source: {}", e);
```